backend/users/signals.py

```
@@ -9,33 +7,14 @@
 @receiver(post_save, sender=User)
 def create_referral_code(sender, instance, created, **kwargs):
     """
     Automatically generate a unique referral code for new users.
     """
     if created or not instance.referral_code:
         try:
-            referral_code = generate_unique_referral_code()
-            # Use update to avoid triggering the signal again
-            User.objects.filter(pk=instance.pk).update(referral_code=referral_code)
+            instance.ensure_referral_code()
         except Exception as e:
```

Ruff (0.15.20)

[warning] Do not catch blind exception: `Exception`

(BLE001)